[Bug]: GetType can still be called as property function, due to case-sensitive comparison

### Issue Description

<https://github.com/dotnet/msbuild/pull/6769> supposedly prevented calling GetType() as a property function, but it does not recognize the name if written with different letter case, e.g. `gettype`.

### Steps to Reproduce

demo.proj:

```XML
<Project>
  <PropertyGroup>
    <foo>aa</foo>
  </PropertyGroup>

  <Target Name="bb">
    <Message Importance="high" Text="$(foo.gettype().FullName)"/>
  </Target>
</Project>
```

dotnet msbuild demo.proj

### Expected Behavior

```
$ dotnet msbuild demo.proj
MSBuild version 17.9.6+a4ecab324 for .NET
demo.proj(7,32): error MSB4185: The function "GetType" on type "System.String" is not available for execution as an MSBuild property function.
```

### Actual Behavior

```
$ dotnet msbuild demo.proj
MSBuild version 17.9.6+a4ecab324 for .NET
  System.String
```

### Analysis

Perhaps a case-insensitive comparison here would fix it: <https://github.com/dotnet/msbuild/blob/a4ecab324c0586fe69b6bdcc062264b244dd8cd0/src/Build/Evaluation/Expander.cs#L5287>

Alternatively, compare the MemberInfo.Name string after the lookup, rather than the user-specified string.  That might be a larger change, though.

### Versions & Configurations

MSBuild version 17.9.6+a4ecab324 in .NET SDK 8.0.202

--- a/src/Build.UnitTests/Evaluation/Expander_Tests.cs
+++ b/src/Build.UnitTests/Evaluation/Expander_Tests.cs
@@ -4978,6 +4978,56 @@ $(
             }
         }
 
+        [Theory]
+        [InlineData("getType")]
+        [InlineData("GetType")]
+        [InlineData("gettype")]
+        public void GetTypeMethod_ShouldNotBeAllowed(string methodName)
+        {
+            using (var env = TestEnvironment.Create())
+            {
+                var root = env.CreateFolder();
+
+                var projectFile = env.CreateFile(root, ".proj",
+                    @$"<Project>
+    <PropertyGroup>
+        <foo>aa</foo>
+        <typeval>$(foo.{methodName}().FullName)</typeval>
+    </PropertyGroup>
+</Project>");
+                var exception = Should.Throw<InvalidProjectFileException>(() =>
+                {
+                    new ProjectInstance(projectFile.Path);
+                });
+                exception.BaseMessage.ShouldContain($"The function \"{methodName}\" on type \"System.String\" is not available for execution as an MSBuild property function.");
+            }
+        }
+
+        [Theory]
+        [InlineData("getType")]
+        [InlineData("GetType")]
+        [InlineData("gettype")]
+        public void GetTypeMethod_ShouldBeAllowed_EnabledByEnvVariable(string methodName)
+        {
+            using (var env = TestEnvironment.Create())
+            {
+                env.SetEnvironmentVariable("MSBUILDENABLEALLPROPERTYFUNCTIONS", "1");
+                var root = env.CreateFolder();
+
+                var projectFile = env.CreateFile(root, ".proj",
+                    @$"<Project>
+    <PropertyGroup>
+        <foo>aa</foo>
+        <typeval>$(foo.{methodName}().FullName)</typeval>
+    </PropertyGroup>
+</Project>");
+                Should.NotThrow(() =>
+                {
+                    new ProjectInstance(projectFile.Path);
+                });
+            }
+        }
+
         /// <summary>
         /// Determines if ICU mode is enabled.
         /// Copied from: https://learn.microsoft.com/en-us/dotnet/core/extensions/globalization-icu#determine-if-your-app-is-using-icu

--- a/src/Build.UnitTests/Evaluation/Expander_Tests.cs
+++ b/src/Build.UnitTests/Evaluation/Expander_Tests.cs
@@ -4984,22 +4984,38 @@ $(
         [InlineData("gettype")]
         public void GetTypeMethod_ShouldNotBeAllowed(string methodName)
         {
-            using (var env = TestEnvironment.Create())
-            {
-                var root = env.CreateFolder();
+            var currentThread = Thread.CurrentThread;
+            var originalCulture = currentThread.CurrentCulture;
+            var originalUICulture = currentThread.CurrentUICulture;
+            var enCultureInfo = new CultureInfo("en");
 
-                var projectFile = env.CreateFile(root, ".proj",
-                    @$"<Project>
-    <PropertyGroup>
-        <foo>aa</foo>
-        <typeval>$(foo.{methodName}().FullName)</typeval>
-    </PropertyGroup>
-</Project>");
-                var exception = Should.Throw<InvalidProjectFileException>(() =>
+            try
+            {
+                currentThread.CurrentCulture = enCultureInfo;
+                currentThread.CurrentUICulture = enCultureInfo;
+
+                using (var env = TestEnvironment.Create())
                 {
-                    new ProjectInstance(projectFile.Path);
-                });
-                exception.BaseMessage.ShouldContain($"The function \"{methodName}\" on type \"System.String\" is not available for execution as an MSBuild property function.");
+                    var root = env.CreateFolder();
+
+                    var projectFile = env.CreateFile(root, ".proj",
+                        @$"<Project>
+            <PropertyGroup>
+                <foo>aa</foo>
+                <typeval>$(foo.{methodName}().FullName)</typeval>
+            </PropertyGroup>
+        </Project>");
+                    var exception = Should.Throw<InvalidProjectFileException>(() =>
+                    {
+                        new ProjectInstance(projectFile.Path);
+                    });
+                    exception.BaseMessage.ShouldContain($"The function \"{methodName}\" on type \"System.String\" is not available for execution as an MSBuild property function.");
+                }
+            }
+            finally
+            {
+                currentThread.CurrentCulture = originalCulture;
+                currentThread.CurrentUICulture = originalUICulture;
             }
         }
 

--- a/src/Build/Evaluation/Expander.cs
+++ b/src/Build/Evaluation/Expander.cs
@@ -5308,7 +5308,7 @@ namespace Microsoft.Build.Evaluation
                 }
 
                 // This could be expanded to an allow / deny list.
-                return methodName != "GetType";
+                return !string.Equals("GetType", methodName, StringComparison.OrdinalIgnoreCase);
             }
 
             /// <summary>

